ios/adapters/epson/NetPrinterEpsonAdapter.m

```diff
@@ -43,58 +36,56 @@ - (void)connectAndSend:(NSString *)host
                success:(RCTResponseSenderBlock)successCallback
                   fail:(RCTResponseSenderBlock)errorCallback {
 
-  _retryAttempts = 3; // TEMP CHANGE NO RETRIES
   _finishedAsyncCall = 0;
   @autoreleasepool {
-    Epos2Printer *printer =
-        [[Epos2Printer alloc] initWithPrinterSeries:modelNumber
-                                               lang:EPOS2_MODEL_ANK];
-    NSString *printerLock;
-    // printerLocksByIp object is a global variable, hence its ref address is
+    Epos2Printer *printer;
+
+    // printersByIP object is a global variable, hence its ref address is
     // being used here as the ID for this code block. Any thread calling this
     // code block of this ID, will lock, or wait until other the lock of
-    // printerLocksByIp is released.
-    if (printerLocksByIp == nil) {
-      @synchronized(printerLocksByIp) {
-        if (printerLocksByIp == nil) {
-          printerLocksByIp = [[NSMutableDictionary alloc] init];
+    // printersByIP is released.
+    if (printersByIP == nil) {
+      @synchronized(printersByIP) {
+        if (printersByIP == nil) {
+          printersByIP = [[NSMutableDictionary alloc] init];
         }
       }
     }
 
-    @synchronized(printerLocksByIp) {
-      printerLock = [printerLocksByIp objectForKey:host];
-      if (printerLock == nil) {
-        printerLock = [NSString stringWithFormat:@"%@", host];
-        [printerLocksByIp setObject:printerLock forKey:host];
+    @synchronized(printersByIP) {
+      printer = [printersByIP objectForKey:host];
+      if (printer == nil) {
+        printer = [[Epos2Printer alloc] initWithPrinterSeries:modelNumber
+                                                         lang:EPOS2_MODEL_ANK];
+
+        [printersByIP setObject:printer forKey:host];
       }
     }
 
-    @synchronized(printerLock) {
+    @synchronized(printer) {
       @try {
         __weak NetPrinterEpsonAdapter *weakSelf = self;
         [printer setReceiveEventDelegate:weakSelf];
         [printer setConnectionEventDelegate:weakSelf];
-        do {
-          @try {
-
-            _retryAttempts++;
-            [self connectAndSendAux:host
-                       printRawData:text
-                            success:successCallback
-                               fail:errorCallback
-                         printerObj:printer];
-            _retryAttempts = 3;
-          } @catch (NSException *exception) {
-            if (_retryAttempts >= 3) {
-              @throw exception;
-            }
-            [printer endTransaction];
-            [self netAdapterDisconnect:printer];
-            [printer clearCommandBuffer];
-            [NSThread sleepForTimeInterval:(3.0f * _retryAttempts)];
+
+          NSString *target = [NSString stringWithFormat:@"TCP:%@", host];
+          int result = EPOS2_SUCCESS;
+          result = [self netAdapterConnect:printer target:target];
+          if (result != EPOS2_SUCCESS) {
+
+            [NSException
```

Comment:
You need to call printer.disconnect when disconnect code is 5

Reply:
This will be done in the catch block